internal/memdb/memdb.go

```diff
@@ -154,17 +161,22 @@ func (c *client) Write(ctx context.Context, tables schema.Tables, resources <-ch
 
 	for resource := range resources {
 		c.memoryDBLock.Lock()
+		resource.Retain()
+		tableName, err := schema.TableNameFromSchema(resource.Schema())
+		if err != nil {
+			return err
+		}
 		if c.spec.WriteMode == specs.WriteModeAppend {
-			c.memoryDB[resource.TableName] = append(c.memoryDB[resource.TableName], resource.Data)
+			c.memoryDB[tableName] = append(c.memoryDB[tableName], resource)
 		} else {
-			c.overwrite(tables.Get(resource.TableName), resource.Data)
+			c.overwrite(tables.Get(tableName), resource)
 		}
 		c.memoryDBLock.Unlock()
 	}
 	return nil
 }
 
-func (c *client) WriteTableBatch(ctx context.Context, table *schema.Table, resources [][]any) error {
+func (c *client) WriteTableBatch(ctx context.Context, table *schema.Table, resources []arrow.Record) error {
```

Comment:
Should this signature use `resources arrow.Record`, rather than `resources []arrow.Record`? I think it's clearer and easier to handle on the destination side if it's a single record (which would contain multiple rows of the same schema)

Reply:
I think we want to enable `[]arrow.Record` i.e both array of records and arrow batches. depending on the source the plugin would group them together that the total records is X. To avoid copying and creating one `record` with all the batches I think it's good to keep it this way for now.

Reply:
Isn't the writing of the record in this case handled by the SDK? I'm thinking the source would send on the results channel, the SDK would then append to the current record builder, and when the batch is full the record will be created and sent to the destination. Not sure why we would need multiple records in this scenario.